src/lib/dnssd/Resolver.h

```diff
@@ -134,6 +147,20 @@ struct DiscoveredNodeData
     Optional<System::Clock::Milliseconds32> GetMrpRetryIntervalIdle() const { return mrpRetryIntervalIdle; }
     Optional<System::Clock::Milliseconds32> GetMrpRetryIntervalActive() const { return mrpRetryIntervalActive; }
 
+    bool IsDeviceTreatedAsSleepy(const ReliableMessageProtocolConfig * defaultMRPConfig) const
```

Comment:
Why does this appear twice?  Would it be possible to share one, singular implementation?

Reply:
Yes, two different classes. Maybe we can revisit this in the future.